### PR TITLE
bump Cairo version to 2.7.0

### PR DESCRIPTION
**Pull request by javra**

### CODE REVIEW

File: Scarb.toml
Comment (milancermak) [inline Scarb.toml:30]: Will remove this, as it shouldn't be part of the package.

Comment (javra) [reply Scarb.toml]: In my opinion it would be nice to have it in there, because then we wouldn't need to patch the wadray repo in the generation of the fv basis. Not needing to patch around in Cairo repos the entire reason I added the compiler option to Cairo and removed the `inline(always)` occurrences (that change has been merged today) in corelib. :disappointed:

Comment (javra) [reply Scarb.toml]: It's just super annoying to tag a branch with the patch along. What's the issue with having in the repo? The profile doesn't get built unless `scarb --profile fv build` is called.

Comment (javra) [reply Scarb.toml]: But fwiw remove it, it's orthogonal to the version bump anyway

Comment (milancermak) [reply Scarb.toml]: Oooh, I see, I didn't realize that about the workflow. Ok, will put it back in.

File: Scarb.toml
Comment (tserg) [inline Scarb.toml:3]: I think we can bump version to `0.4.0`.

**Review by tserg — APPROVED** [review]
LGTM, just one small nit to bump the package version in preparation for release.

**Review by tserg — COMMENTED** [review]
It just came to my mind, should we set the edition here like in access control library too?

**Comment by tserg** [comment]
> I think wadray should be independent of the edition, so we don't need to specify it. I'm also pretty sure that if it's just imported by another project, the compiler will use the edition of the importing package anyway.

Sounds ok to me, I will defer to @milancermak to take a final look before merging.

**Comment by milancermak** [comment]
> I'm also pretty sure that if it's just imported by another project, the compiler will use the edition of the importing package anyway

Wow, will it? TBH I'm have no idea how the editions work when importing, but this seems strange to me. I have the commit to update to 2024_07 edition ready locally, most of the updates there are about adding `pub` but also changes to import paths - some crates moved, some stuff became private and/or accessible via a simple `into()`.

That's why it's strange to me that if this lib (which implicitly has edition 2023_01) would be imported into a 2024_07 project, it would not produce any errors. Do you know how that works? Does the compiler compile the dependencies separately using whatever is in their Scarb.toml?

Re: version change - I didn't want to do that just yet as I wanted to add a `wad!` and `ray!` macros to the library (and also do the 2024_07 edition 😬) and do a release only then.

**Comment by javra** [comment]
> That's why it's strange to me that if this lib (which implicitly has edition 2023_01) would be imported into a 2024_07 project, it would not produce any errors. Do you know how that works? Does the compiler compile the dependencies separately using whatever is in their Scarb.toml?

Just asked in the TG and apparently the compiler does compile each crate with its own edition, so I was wrong above.

**Comment by milancermak** [comment]
Saw the discussion, thanks for checking!

Will unbump the version then, so I can sneak some more changes before 0.4.0 :)

**Pull request merged**
